### PR TITLE
Refactor/upgrade to w3mv2

### PR DESCRIPTION
**Pull request by devceline**
# Description

- Upgrade to web3modal v2.5
- Remove `get-ens`
- Add new method for fetching ens records 
- Fix bugs with previous ens fetching
- Remove `ethers` and add `viem` for `wagmi` to work with web3modal v2.5

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by bkrem** [comment]
Just tried to test via the preview, still seems to be providing v1 pairingURIs: `wc:c5be90ad-7c28-4f59-b730-079ee48cc859@1?bridge=https%3A%2F%2Fa.bridge.walletconnect.org&key=4c84d1edf56dda9d108c1d13bd7968fd5c1c648b8cf1a0a54ad8717299e281c9`

**Comment by devceline** [comment]
> Just tried to test via the preview, still seems to be providing v1 pairingURIs: `wc:c5be90ad-7c28-4f59-b730-079ee48cc859@1?bridge=https%3A%2F%2Fa.bridge.walletconnect.org&key=4c84d1edf56dda9d108c1d13bd7968fd5c1c648b8cf1a0a54ad8717299e281c9`

I was going to do that in a separate PR, but this is a good place to do it :D

**Comment by bkrem** [comment]
Did some dogfooding with the v2 flow, looks like there are still some bits to be smoothed out:

* Seeing `Error: No matching key. pairing: undefined` on modal init (I think Gancho just patched this upstream though)
* After providing the second signature I just seem to be stuck on the 2nd step signing modal (see screenshot). Tried with both Spot and our iOS TF WalletApp
* On logout I get the following keychain errors:

```
Error: No matching key. keychain: 0520953cab5c2536e24ba20bd25be43b0b3298f6cc4b9075c06da66dfe70b610
    at y_.get (index.es.90eab585.js:7:15494)
    at m_.getSymKey (index.es.90eab585.js:7:18196)
    at m_.encode (index.es.90eab585.js:7:17280)
    at X_.sendRequest (index.es.90eab585.js:7:66184)
    at X_.disconnect (index.es.90eab585.js:7:63955)
    at async Sa.disconnect (index.es.90eab585.js:7:82876)
    at async Pa.disconnect (index.es.90eab585.js:33:26917)
    at async Na.disconnect (index.es.90eab585.js:33:36528)
    at async S3e.connect (index.b3e9ee9c.js:31:16790)
    at async aU (index.b3e9ee9c.js:31:7440)

...

Error: No matching key. keychain: 0520953cab5c2536e24ba20bd25be43b0b3298f6cc4b9075c06da66dfe70b610
    at y_.get (index.es.90eab585.js:7:15494)
    at m_.getSymKey (index.es.90eab585.js:7:18196)
    at m_.encode (index.es.90eab585.js:7:17280)
    at X_.sendRequest (index.es.90eab585.js:7:66184)
    at X_.disconnect (index.es.90eab585.js:7:63955)
    at async Sa.disconnect (index.es.90eab585.js:7:82876)
    at async Pa.disconnect (index.es.90eab585.js:33:26917)
    at async Na.disconnect (index.es.90eab585.js:33:36528)
    at async S3e.connect (index.b3e9ee9c.js:31:16790)
    at async aU (index.b3e9ee9c.js:31:7440)
```

---

Screenshot 2023-06-05 at 12 17 33

**Comment by bkrem** [comment]
Still bumping into these, only show stopper is the getting stuck after 2nd signature though:

> Did some dogfooding with the v2 flow, looks like there are still some bits to be smoothed out:
> 
> * Seeing `Error: No matching key. pairing: undefined` on modal init (I think Gancho just patched this upstream though)
> * After providing the second signature I just seem to be stuck on the 2nd step signing modal (see screenshot). Tried with both Spot and our iOS TF WalletApp
> * On logout I get the following keychain errors:
> 
> ```
> Error: No matching key. keychain: 0520953cab5c2536e24ba20bd25be43b0b3298f6cc4b9075c06da66dfe70b610
>     at y_.get (index.es.90eab585.js:7:15494)
>     at m_.getSymKey (index.es.90eab585.js:7:18196)
>     at m_.encode (index.es.90eab585.js:7:17280)
>     at X_.sendRequest (index.es.90eab585.js:7:66184)
>     at X_.disconnect (index.es.90eab585.js:7:63955)
>     at async Sa.disconnect (index.es.90eab585.js:7:82876)
>     at async Pa.disconnect (index.es.90eab585.js:33:26917)
>     at async Na.disconnect (index.es.90eab585.js:33:36528)
>     at async S3e.connect (index.b3e9ee9c.js:31:16790)
>     at async aU (index.b3e9ee9c.js:31:7440)
> 
> ...
> 
> Error: No matching key. keychain: 0520953cab5c2536e24ba20bd25be43b0b3298f6cc4b9075c06da66dfe70b610
>     at y_.get (index.es.90eab585.js:7:15494)
>     at m_.getSymKey (index.es.90eab585.js:7:18196)
>     at m_.encode (index.es.90eab585.js:7:17280)
>     at X_.sendRequest (index.es.90eab585.js:7:66184)
>     at X_.disconnect (index.es.90eab585.js:7:63955)
>     at async Sa.disconnect (index.es.90eab585.js:7:82876)
>     at async Pa.disconnect (index.es.90eab585.js:33:26917)
>     at async Na.disconnect (index.es.90eab585.js:33:36528)
>     at async S3e.connect (index.b3e9ee9c.js:31:16790)
>     at async aU (index.b3e9ee9c.js:31:7440)
> ```
> 
> Screenshot 2023-06-05 at 12 17 33

---

If I manually refresh the page after getting stuck it loads the w3i UI, but I see:
```
{context: 'syncClient/sync_com.walletconnect.chat.inviteKeys'} 'No matching key. com.walletconnect.chat.inviteKeys: eip155:1:0x3d3429CEe7af5884618079aBfCD2566f67CefBc3'
```

File: src/contexts/W3iContext/hooks/chatHooks.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export const useChatState = (w3iProxy: Web3InboxProxy, proxyReady: boolean) => {
       if (chatClient && key && uiEnabled.chat) {
         try {
           const registeredKeyRes = await chatClient.register({ account: `eip155:1:${key}` })
+          console.log({ registeredKeyRes })
```
Comment (bkrem) [inline src/contexts/W3iContext/hooks/chatHooks.ts:52]: Needed?

**Review by bkrem — APPROVED** [review]
LGTM, also just tested the preview and logged in smoothly 💪

**Pull request merged**